File: Official Windows Platform Sample/Windows Phone 8.1 samples/[C++]-Windows Phone 8.1 samples/Contact Picker app sample/C++/WindowsPhone/PhoneConfiguration.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include <ppltasks.h>
#include <ppl.h>  
#include "PhoneConfiguration.h"

using namespace concurrency;
using namespace Platform;
using namespace Windows::Phone::PersonalInformation;
using namespace Windows::Foundation;
using namespace Windows::Foundation::Collections;
using namespace SDKSample;

/// <summary>
/// Creates a contact store and add contacts.
/// </summary>
void PhoneConfiguration::CreateContactStore()
{
	create_task(ContactStore::CreateOrOpenAsync(ContactStoreSystemAccessMode::ReadWrite,
		ContactStoreApplicationAccessMode::ReadOnly)).then([](ContactStore^ contactStore)
	{
		for (SampleContact sampleContact : ContactCreator::CreateSampleContacts())
		{
			StoredContact^ contact = ref new StoredContact(contactStore);
			create_task(contact->GetPropertiesAsync()).then([sampleContact](IMap <String^, Object^>^ props)
			{
				if (!(sampleContact.FirstName->IsEmpty()))
				{
					props->Insert(KnownContactProperties::GivenName, sampleContact.FirstName);
				}
				
				if (!sampleContact.HomeEmail->IsEmpty())
				{
					props->Insert(KnownContactProperties::Email, sampleContact.HomeEmail);
				}

				if (!sampleContact.LastName->IsEmpty())
				{
					props->Insert(KnownContactProperties::FamilyName, sampleContact.LastName);
				}

				if (!sampleContact.HomeEmail->IsEmpty())
				{
					props->Insert(KnownContactProperties::Email, sampleContact.HomeEmail);
				}

				if (!sampleContact.WorkEmail->IsEmpty())
				{
					props->Insert(KnownContactProperties::WorkEmail, sampleContact.WorkEmail);
				}

				if (!sampleContact.HomePhone->IsEmpty())
				{
					props->Insert(KnownContactProperties::Telephone, sampleContact.HomePhone);
				}

				if (!sampleContact.WorkPhone->IsEmpty())
				{
				    props->Insert(KnownContactProperties::CompanyTelephone, sampleContact.WorkPhone);
				}

				if (!sampleContact.MobilePhone->IsEmpty())
				{
					props->Insert(KnownContactProperties::MobileTelephone, sampleContact.MobilePhone);
				}
			}).then([contact]()
			{
				contact->SaveAsync();
			});
		}
	});
}

File: Official Windows Platform Sample/Windows 8.1 Store app samples/[C#]-Windows 8.1 Store app samples/Contact Picker app sample/C#/WindowsPhone/PhoneConfiguration.cs
// Copyright (c) Microsoft. All rights reserved.

using System;
using System.Collections.Generic;
using Windows.Phone.PersonalInformation;

namespace SDKTemplate
{
    /// <summary>
    /// Sets phone configuration.
    /// </summary>
    public static class PhoneConfiguration
    {
        /// <summary>
        /// Creates a contact store and add contacts.
        /// </summary>
        public static async void CreateContactStore()
        {
            ContactStore contactStore = await ContactStore.CreateOrOpenAsync(
                            ContactStoreSystemAccessMode.ReadWrite,
                            ContactStoreApplicationAccessMode.ReadOnly);

            foreach (SampleContact sampleContact in SampleContact.CreateSampleContacts())
            {
                StoredContact contact = new StoredContact(contactStore);
                IDictionary<string, object> props = await contact.GetPropertiesAsync();

                if (!string.IsNullOrEmpty(sampleContact.FirstName))
                {
                    props.Add(KnownContactProperties.GivenName, sampleContact.FirstName);
                }

                if (!string.IsNullOrEmpty(sampleContact.LastName))
                {
                    props.Add(KnownContactProperties.FamilyName, sampleContact.LastName);
                }

                if (!string.IsNullOrEmpty(sampleContact.HomeEmail))
                {
                    props.Add(KnownContactProperties.Email, sampleContact.HomeEmail);
                }

                if (!string.IsNullOrEmpty(sampleContact.WorkEmail))
                {
                    props.Add(KnownContactProperties.WorkEmail, sampleContact.WorkEmail);
                }

                if (!string.IsNullOrEmpty(sampleContact.HomePhone))
                {
                    props.Add(KnownContactProperties.Telephone, sampleContact.HomePhone);
                }

                if (!string.IsNullOrEmpty(sampleContact.WorkPhone))
                {
                    props.Add(KnownContactProperties.CompanyTelephone, sampleContact.WorkPhone);
                }

                if (!string.IsNullOrEmpty(sampleContact.MobilePhone))
                {
                    props.Add(KnownContactProperties.MobileTelephone, sampleContact.MobilePhone);
                }

                await contact.SaveAsync();
            }
        }
    }
}